lib/ansible/modules/windows/win_git.ps1

```diff
@@ -0,0 +1,201 @@
+        dest              = $null
+        replace_dest      = $false
+        accept_hostkey    = $false
+    }
+    msg = $null
+    changed = $false
+}
+
+$name = Get-AnsibleParam -obj $params -name "name" -failifempty $true -type "str"
+$dest = Get-AnsibleParam -obj $params -name "dest" -failifempty $true -type "path"
+$replace_dest = Get-AnsibleParam -obj $params -name "replace_dest" -default $false -type "bool"
+$accept_hostkey = Get-AnsibleParam -obj $params -name "accept_hostkey" -default $false -type "bool"
+$check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -default $false -type "bool"
+
+# Add Git to PATH variable
+$env:Path += ";" + "C:\Program Files\Git\usr\bin"
+$env:Path += ";" + "C:\Program Files (x86)\Git\usr\bin"
+
+# Functions
+Function Find-Command
+{
+    [CmdletBinding()]
+    param(
+      [Parameter(Mandatory=$true, Position=0)] [string] $command
+    )
+    $installed = get-command $command -erroraction Ignore
+    write-verbose "$installed"
+    if ($installed)
+    {
+        return $installed
+    }
+    return $null
+}
+
+Function FindGit
+{
+    [CmdletBinding()]
+    param()
+    $p = Find-Command "git.exe"
+    if ($p -ne $null)
+    {
+        return $p
+    }
+    $a = Find-Command "C:\Program Files\Git\bin\git.exe"
+    if ($a -ne $null)
+    {
+        return $a
+    }
+    Throw "git.exe is not installed. It must be installed (use chocolatey)"
+}
+
+# Check destination folder, create if not exist
+function PrepareDestination
+{
+    [CmdletBinding()]
+    param()
+    if (Test-Path $dest) {
+        $directoryInfo = Get-ChildItem $dest -Force | Measure-Object
+        if ($directoryInfo.Count -ne 0) {
+            if ($replace_dest) {
+                # Clean destination
+                Remove-Item $dest -Force -Recurse | Out-Null
+            }
+            else {
+                Throw "Destination folder not empty!"
+            }
+        }
+    }
+    else {
+        # Create destination folder
+        New-Item $dest -ItemType Directory -Force | Out-Null
+    }
+}
+
+# SSH Keys
+function CheckSshKnownHosts
+{
+    [CmdletBinding()]
+    param()
+    # Get the Git Hostname
+    $gitServer = $($name -replace "^(\w+)\@([\w-_\.]+)\:(.*)$", '$2')
+    & cmd /c ssh-keygen.exe -F $gitServer | Out-Null
+    $rc = $LASTEXITCODE
+    
+    if ($rc -ne 0)
+    {
+        # Host is unknown
+        if ($accept_hostkey)
+        {
+            & cmd /c ssh-keyscan.exe -t ecdsa-sha2-nistp256 $gitServer | Out-File -Append "$env:Userprofile\.ssh\known_hosts"
+        }
+        else
+        {
+            Trow "Host is not known!"
+        }
+    }
+}
+
+function CheckSshIdentity
+{
+    [CmdletBinding()]
+    param()
+
+    & cmd /c git.exe ls-remote $name | Out-Null
+    $rc = $LASTEXITCODE
+    if ($rc -ne 0) {
+        Trow "Something wrong with connection!"
+    }
+}
+
+# Build Arguments
+$git_opts = @()
+
+$git_opts += "clone"
+
+if ($name -eq ($null -or "")) {
```

Comment:
yeah `($null -or "")` will always be false so this turns into
```PowerShell
if ($name -eq $false) 
```
is that expected?  Doesn't seem like the right check
```PowerShell
Phil@PHIL-PC {C:\} $ ($null -or "")
False
Phil@PHIL-PC {C:\} $ $name = ""
Phil@PHIL-PC {C:\} $ $name -eq ($null -or "")
False
Phil@PHIL-PC {C:\} $ $name = $null
Phil@PHIL-PC {C:\} $ $name -eq ($null -or "")
False
Phil@PHIL-PC {C:\} $ $name = $false
Phil@PHIL-PC {C:\} $ $name -eq ($null -or "")
True
```